File: 1.二维数组中的查找/二维数组中的查找.java
public class Solution {
    public boolean Find(int target, int [][] array) {
        int rows = array.length;
        int cols = array[0].length;
        if(rows == 0||cols == 0){
            return false;
        }
        int i = rows-1;
        int j = 0;
        while(i>=0&&j<cols){
            if(array[i][j] > target){
                i--;
            }else if(array[i][j] < target){
                j++;
            }else{
                return true;
            }
        }
        return false;
    }
}

File: 1.二维数组中的查找/二维数组中的查找.cpp
class Solution {
public:
    bool Find(int target, vector<vector<int> > array) {
        int m = array.size();
        if(m == 0){
            return false;
        }
        int n = array[0].size();
        if(n == 0){
            return false;
        }
        for(int i = m-1,j = 0; i >= 0&&j < n;){
            if(array[i][j] > target){
                i--;
            }else if(array[i][j] < target){
                j++;
            }else{
                return true;
            }
        }
        return false;
    }
};
